### ISSUE TITLE
Mod description with single apostrophe leads to "Illegal format found" error in log

### ISSUE THREAD

**Issue by 3TUSK** [neoforged/NeoForge#1561]
**Minecraft Version:** 1.20.5 - 1.21.1. Tested using 1.21.1.

**NeoForge Version:** Tested using 21.1.61; should happen on any NeoForge with version in range `[20.5.0, 21.1.61]`

**Logs:** N/A, no crashes happen

**Steps to Reproduce:**
 1. Create a low-code mod with only `neoforge.mods.toml` given below
```toml
modLoader="lowcodefml"
loaderVersion="[1,)"
license="CC0-1.0"

[[mods]] #mandatory
modId="test_description_with_single_quote"
version="1.0.0"
displayName="Test Description with Single Quote" 
description="It's regression time!"
```
 2. Add it to your `mods` directory
 3. Launch game
 4. Open Mod List screen
 5. Confirm that the low code mod is loaded by finding it on your left-hand side
 6. Click the low code mod entry to open details
 7. Find this line in your log, timestamp may vary: 
```
[18:46:26] [Render thread/ERROR] [ne.ne.fm.i1.FMLTranslations/CORE]: Illegal format found `It's regression time!`
```

**Description of issue:**

Similar to #734, if mod description contains an apostrophe (U+0027 `APOSTROPHE`), `ExtendedMessageFormat`, which is used by `FMLTranslations`, will see it as unterminated quoted string and throw `IllegalArgumentException`. 

It was fixed in #686 for 1.20.4, but the fix for 1.20.5 and onward - #851 - only covers mod name, not mod description. 

This regression was attested in https://github.com/Argon4W/EveryXHotpot/commit/bb5458cfa9b1e7eeeb0e7745fa4295d80cde4019.

**Comment by Corail31** [comment]
It's still a problem even in 1.21.3.

**Comment by costentz** [comment]
The issue is still present in the 21.1.92 version of Neoforge. Will the fix be ported to 1.21.1?
Every time a mod with a long description is selected from the menu the log gets filled with its content.
